python/LAMBDA_CONCURRENCY_CHECK/LAMBDA_CONCURRENCY_CHECK.py

```diff
@@ -0,0 +1,476 @@
+     Then: Return an Error
+  Scenario: 3
+    Given: Lambda function is not configured for Concurrency i.e. Concurrency is not set
+     Then: Return NON_COMPLIANT
+  Scenario: 4
+    Given: Lambda function is configured for Concurrency i.e. Concurrency is set
+      And: the ConcurrencyLimitLow parameter is not configured
+      And: the ConcurrencyLimitHigh parameter is not configured
+     Then: Return COMPLIANT
+  Scenario: 5
+    Given: Lambda function is configured for Concurrency i.e. Concurrency is set
+      And: the ConcurrencyLimitLow parameter is configured and valid
+      And: the function concurrency < ConcurrencyLimitLow
+     Then: Return NON_COMPLIANT
+  Scenario: 6
+    Given: Lambda function is configured for Concurrency i.e. Concurrency is set
+      And: the ConcurrencyLimitLow parameter is configured and valid
+      And: the function concurrency ≥ ConcurrencyLimitLow
+      And: the ConcurrencyLimitHigh parameter is not configured
+     Then: Return COMPLIANT
+  Scenario: 7
+    Given: Lambda function is configured for Concurrency i.e. Concurrency is set
+      And: the ConcurrencyLimitHigh parameter is configured and valid
+      And: the function concurrency > ConcurrencyLimitHigh
+     Then: Return NON_COMPLIANT
+  Scenario: 8
+    Given: Lambda function is configured for Concurrency i.e. Concurrency is set
+      And: the ConcurrencyLimitHigh parameter is configured and valid
+      And: the function concurrency ≤ ConcurrencyLimitHigh
+      And: the ConcurrencyLimitLow parameter is not configured
+     Then: Return COMPLIANT
+  Scenario: 9
+    Given: Lambda function is configured for Concurrency i.e. Concurrency is set
+      And: the ConcurrencyLimitLow parameter is configured and valid
+      And: the function concurrency ≥ ConcurrencyLimitLow
+      And: the ConcurrencyLimitHigh parameter is configured and valid
+      And: the function concurrency ≤ ConcurrencyLimitHigh
+     Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::Lambda::Function'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+
+    function_name = configuration_item['configuration']['functionName']
+    if 'Concurrency' not in configuration_item['supplementaryConfiguration']:
+        return build_evaluation_from_config_item(
+            configuration_item,
+            'NON_COMPLIANT',
+            annotation='Concurrency not set for the lambda function: {}'.format(function_name)

```

Comment:
The function_name is already displayed in the resource id. No need to mention the functino name in the annotations

Reply:
Done!